webtorrent/transport.go

```diff
@@ -39,16 +39,35 @@ type wrappedPeerConnection struct {
 	pproffd.CloseWrapper
 	span trace.Span
 	ctx  context.Context
+
+	onCloseHandler func()
 }
 
 func (me *wrappedPeerConnection) Close() error {
 	me.closeMu.Lock()
 	defer me.closeMu.Unlock()
+
+	me.onClose()
```

Comment:
Do you need to guard against multiple closes here?

Reply:
I'm not super confident with locks, but my understanding is that here `me.onClose()` is called while `me` is still locked, right? I've noticed that if I call it outside the lock, we end up trying to access the stats after the connection is already closed, so we only get `nil` values. But I'm open to suggestions!

Reply:
Ah no I actually mean repeat closes. Just a sync.Once to wrap it perhaps.